modules/core/src/main/java/org/apache/ignite/internal/processors/cache/verify/IdleVerifyUtility.java

```diff
@@ -87,28 +95,131 @@ public static void checkPartitionsPageCrcSum(
         for (int pageNo = 0; pageNo < pageStore.pages(); pageId++, pageNo++) {
             buf.clear();
 
-            if (cpFlag.get())
-                throw new GridNotIdleException(CLUSTER_NOT_IDLE_MSG);
-
             pageStore.read(pageId, buf, true);
         }
+    }
+
+    /**
+     * Gather updateCounters info.
+     *
+     * @param ign Ignite instance.
+     * @param grpIds Group Id`s.
+     * @return Current groups distribution with update counters per partitions.
+     */
+    public static Map<Integer, Map<Integer, Long>> getUpdateCountersSnapshot(
+        IgniteEx ign,
+        Set<Integer> grpIds
+    ) {
+        Map<Integer, Map<Integer, Long>> partsWithCountersPerGrp = new HashMap<>();
+
+        for (Integer grpId : grpIds) {
+            CacheGroupContext grpCtx = ign.context().cache().cacheGroup(grpId);
+
+            if (grpCtx == null)
+                throw new GridNotIdleException("Group not found: " + grpId + "."
+                        + " Possible reasons: rebalance in progress or concurrent cache destroy.");
+
+            GridDhtPartitionTopology top = grpCtx.topology();
+
+            Map<Integer, Long> partsWithCounters =
+                partsWithCountersPerGrp.computeIfAbsent(grpId, k -> new HashMap<>());
+
+            for (GridDhtLocalPartition part : top.currentLocalPartitions()) {
+                if (part.state() != GridDhtPartitionState.OWNING)
+                    continue;
+
+                if (!part.dataStore().partUpdateCounter().sequential())
+                    throw new GridNotIdleException("Cluster not idle. Probably due to continues rebalance");
```

Comment:
Due to lost updates (should not happen under normal circumstances) counter may not be sequential even if grid is idle. Better to compare whole counter before and after check.